include/erlcloud_waf.hrl

```diff
@@ -0,0 +1,126 @@
+-include("erlcloud.hrl").
+
+-type waf_update_action() :: insert | delete.
+-type waf_text_transformation() :: none | compress_white_space | html_entity_decode | lowercase | cmd_line | url_decode.
+-type waf_positional_constraint() :: exactly | starts_with | ends_with | contains | contains_word.
+-type waf_comparison_operator() :: eq | ne | le | lt | ge | gt.
+-type waf_predicate_type() :: ip_match | byte_match | sql_injection_match | size_constraint.
+-type waf_acl_action_type() :: block | allow | count.
+
+-record(waf_field_to_match, {
+    type :: uri | 'query_string' | header | method | body,
+    data :: string()
+}).
```

Comment:
It might be pretty clear to use simple tuples instead of complex records.
Just like we did in `erlcloud_ddb2`:
https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_ddb2.erl#L2051-L2055